packages/flutter/test/widgets/selectable_region_context_menu_test.dart

```diff
@@ -5,40 +5,61 @@
 @TestOn('browser') // This file contains web-only library.
 library;
 
-import 'dart:html' as html;
+import 'dart:js_interop';
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter_test/flutter_test.dart';
+import 'package:web/web.dart' as web;
+
+extension on JSString {
+  external JSBoolean includes(JSString searchString);
+}
+extension on web.HTMLCollection {
+  Iterable<web.Element> get iterable => _genIterable(length, (JSNumber i) => item(i));
+}
+extension on web.CSSRuleList {
+  Iterable<web.CSSRule> get iterable => _genIterable(length, (JSNumber i) => item(i));
+}
+
+typedef ItemGetter<T> = T? Function(JSNumber index);
+Iterable<T> _genIterable<T>(JSNumber length, ItemGetter<T> getItem) {
+  return Iterable<T>.generate(length.toDart.toInt(), (int index) => getItem(index.toJS)!);
+}
 
 void main() {
-  html.Element? element;
+  web.HTMLElement? element;
   PlatformSelectableRegionContextMenu.debugOverrideRegisterViewFactory = (String viewType, Object Function(int viewId) fn, {bool isVisible = true}) {
-    element = fn(0) as html.Element;
+    element = fn(0) as web.HTMLElement;
     // The element needs to be attached to the document body to receive mouse
     // events.
-    html.document.body!.append(element!);
+    web.document.body!.append(element);
   };
   // This force register the dom element.
   PlatformSelectableRegionContextMenu(child: const Placeholder());
   PlatformSelectableRegionContextMenu.debugOverrideRegisterViewFactory = null;
 
   test('DOM element is set up correctly', () async {
     expect(element, isNotNull);
-    expect(element!.style.width, '100%');
-    expect(element!.style.height, '100%');
-    expect(element!.classes.length, 1);
-    final String className = element!.classes.first;
+    expect(element!.style.width, '100%'.toJS);
```

Comment:
@joshualitt On dart2wasm, would it be cheaper to internalize the JS string and then do a Dart equality check versus externalizing a Dart string and then doing a JS equality check? On the JS backends, it's all the same afaik.

Reply:
> On the JS backends, it's all the same afaik.

Right, to my knowledge, `toJS` is a noop for this case in JS backends.

> On dart2wasm, would it be cheaper to internalize the JS string and then do a Dart equality check versus externalizing a Dart string and then doing a JS equality check?

@eyebrowsoffire brought up a good point yesterday that `'100%'.toJS` is easier to optimize in the future to avoid the conversion from Dart string to JS string at runtime.

That said, this is a test which means performance isn't critical. But it would be nice to know the performance tradeoffs for when we start writing code like this for production.

Reply:
Yeah I've talked for a while about adding a kernel transform in dart2wasm that finds constant string expressions with `.toJS` at the end of them and just exporting those from the JS runtime and importing them directly, which avoids any sort of copying at runtime. So I think it's better to do the conversion on the constant expression, because it's much more feasible to optimize that away.

Reply:
Yeah, that makes sense for constant expressions once we have that optimization in place. I do wonder if it's cheaper to avoid the JS `==` call in favor of the Dart `==` call in dart2wasm for cases where we have non-constant Strings, but it's probably similar performance.

Reply:
I think I would actually expect the JS `==`/`===` call to be faster than the dart `==`, assuming we're having some call to javascript either way. V8 probably can use some bulk memory comparison operations like `strcmp` or `memcmp` or something like that in their native JS string comparisons, whereas with dart strings are calling `charCodeAt` and directly comparing each individual code unit. Maybe V8 can JIT that into a bulk memory operation but it seems unclear to me.